rpc/src/rpc.rs

```diff
@@ -1002,26 +1002,37 @@ impl JsonRpcRequestProcessor {
         })
     }
 
-    fn check_blockstore_root<T>(
+    // Check if the given `slot` is within the blockstore bounds. This function assumes that
+    // `result` is from a blockstore fetch, and that the fetch:
+    // 1) Checked if `slot` is above the lowest cleanup slot (and errored if not)
+    // 2) Checked if `slot` is a root
+    fn check_blockstore_bounds<T>(
         &self,
         result: &std::result::Result<T, BlockstoreError>,
         slot: Slot,
     ) -> Result<()> {
-        if let Err(err) = result {
-            debug!(
-                "check_blockstore_root, slot: {:?}, max root: {:?}, err: {:?}",
-                slot,
-                self.blockstore.max_root(),
-                err
-            );
-            if slot >= self.blockstore.max_root() {
-                return Err(RpcCustomError::BlockNotAvailable { slot }.into());
-            }
-            if self.blockstore.is_skipped(slot) {
-                return Err(RpcCustomError::SlotSkipped { slot }.into());
+        match result {
+            // The slot was found, all good
+            Ok(_) => Ok(()),
+            // The slot was cleaned up, return Ok() for now to allow fallback to bigtable
+            Err(BlockstoreError::SlotCleanedUp) => Ok(()),
+            // The slot was not cleaned up but also not found
+            Err(BlockstoreError::SlotNotRooted) => {
+                let max_root = self.blockstore.max_root();
+                debug!("check_blockstore_bounds, slot: {slot}, max root: {max_root}");
+                // Our node hasn't seen this slot yet, error out
+                if slot >= max_root {
```

Comment:
@CriesofCarrots - I went ahead and removed `Blockstore::is_skipped()`. However, one other item for discussion in this PR is this check here.

What do you think of moving this check into `Blockstore::get_rooted_*` functions? I'm thinking the flow would look like:
```rust
        if self.is_root(slot) {
            self.blocktime_cf
                .get(slot)?
                .ok_or(BlockstoreError::SlotUnavailable)
        } else {
            if slot > self.max_root() {
                Err(BlockstoreError::SlotUnavailable)
            } else {
                Err(BlockstoreError::SlotNotRooted)
            }
        }

I think this would be nice because all of the bounds checking logic would be in the same place. With the move to Blockstore, the function in `rpc.rs` would be a simple mapper from `BlockstoreError` to `RpcCustomError`

Reply:
> What do you think of moving this check into `Blockstore::get_rooted_*` functions?

> I think this would be nice because all of the bounds checking logic would be in the same place.

Yes, that makes sense to me. It's also consistent with how we already handle cleaned up slots -- actual check done in Blockstore, then interpret error to decide on handling in RPC (fall back to Bigtable, etc).

I think the Skipped rework in this PR is still needed, so it probably makes sense to merge this and then do that change in a separate PR. I'll approve so you can keep moving if you agree.

>  Thinking the checks being in `rpc.rs` might have just been a result of how the API involved from inception to where it is now

Yes, totally. It wasn't an intentional design 😞

Reply:
> I think the Skipped rework in this PR is still needed, so it probably makes sense to merge this and then do that change in a separate PR. I'll approve so you can keep moving if you agree.

I'm cool with that - keep the individual changes easier to reason about and review.

> Yes, totally. It wasn't an intentional design 😞

Definitely not pointing fingers; I think plenty of more pressing things have prevented this code from getting TLC prior to our recent foray